### PR TITLE
feat: request observers to intercept requests and responses

### PR DESCRIPTION
**Pull request by arifBurakDemiray**
This is needed for module content's fetching mechanism

### CODE REVIEW

File: sdk/src/main/java/ly/count/android/sdk/ConnectionProcessor.java [hunk 1]
```diff
@@ -494,6 +502,10 @@ public void run() {
                                 L.v("[ConnectionProcessor] Response was a unknown, will retry");
                                 rRes = RequestResult.RETRY;
                             } else {
+                                // notify observers for the response
```
Comment (arifBurakDemiray) [inline sdk/src/main/java/ly/count/android/sdk/ConnectionProcessor.java:505]: add tests for this

**Comment by arifBurakDemiray** [comment]
also fix codacy

**Comment by turtledreams** [comment]
le conflic

**Comment by arifBurakDemiray** [comment]
we might omit this PR for now, this was needed for content update from RQ but we are not doing that anymore

**Pull request closed without merging**